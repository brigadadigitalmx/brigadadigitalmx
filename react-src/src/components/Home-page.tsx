import React, { Component} from 'react';

import { Header, Footer } from './common';
import { PageSection } from './sections';

export class HomePage extends Component<{}, {data: [] }> {
  constructor(props){
    super(props);
    this.state = {
      data : []
    };
    
  }
  
  
  async componentDidMount(){
    return fetch('/wp-json/wp/v2/pages?slug=homepage')
    .then(response => response.json())
    .then(responseJson => fetch(`/wp-json/wp/v2/pages?parent=${responseJson[0].id}&orderBy=date&order=asc`))
    .then(response => response.json())
    .then(responseJson => this.setState({data: responseJson}))
    .catch(e => console.error(e));
  }

  componentDidCatch(error, errorInfo) { 
    // You can also log the error to an error reporting service    logErrorToMyService(error, errorInfo);
    console.error(error, errorInfo);
   }  
  
  render(){
    return (
      <main>
      <Header collapse={false} />
      {this.state.data.map((item : any, key) => {
        return (
          <PageSection key={key} theme={key % 2 === 0 ? 'light' : 'dark'} classes="padding-top-three padding-bottom-three">
            <h1 className="cormorant big"> {item.title.rendered} </h1>
            <div dangerouslySetInnerHTML={{ __html: item.content.rendered }} />
          </PageSection>
        )
      })}
      <Footer theme="dark" classes="padding-top-three" />
      </main>
    )
  }
}

export default HomePage;