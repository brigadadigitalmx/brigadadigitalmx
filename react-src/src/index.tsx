import React from 'react';
import ReactDOM from 'react-dom';
import * as serviceWorker from './serviceWorker';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { HomePage, Faqs, Us, Projects } from './components';

export default function MainRouter() {
  return (
    <React.StrictMode>
    <BrowserRouter>
    <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/faqs" component={Faqs} />
        <Route exact path="/us" component={Us} />
        <Route exact path="/projects" component={Projects} />
      </Switch>
    </BrowserRouter>
    </React.StrictMode>
  );
}

ReactDOM.render(
  <MainRouter />,
  document.getElementById('main')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();
