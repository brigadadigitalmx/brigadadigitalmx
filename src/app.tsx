import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import { RouterStore, syncHistoryWithStore } from 'mobx-react-router';
import { Provider } from 'mobx-react';

import { HomePage } from './components';
import './less/main.less';

const browserHistory = createBrowserHistory();
const routingStore = new RouterStore();
const stores = {
  routing: routingStore,
};

const history = syncHistoryWithStore(browserHistory, routingStore);

export default function MainRouter() {
  return (
    <BrowserRouter>
      <Provider {...stores}>
        <Router history={history}>
          <Switch>
            <Route exact path="/" component={HomePage} />
          </Switch>
        </Router>
      </Provider>
    </BrowserRouter>
  );
}

ReactDOM.render(<MainRouter />, document.getElementById('main'));
