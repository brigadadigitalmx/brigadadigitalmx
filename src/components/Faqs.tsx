import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';

import { Header, Footer } from './common';
import { PageSection } from './sections';
import { ListedFaq } from './text';

export const Faqs = () => {
  return (
    <main>
      <Header appearance="extra-collapsed" />
      <PageSection theme="light" classes="padding-top-three padding-bottom-three">
        <h1 className="cormorant big">Gobiernos</h1>
        <div className="margin-top-one padding-left-one">
          <ListedFaq
            question="¿Cómo solicito un voluntario?"
            answer="Complete este formulario y comparta la mayor cantidad de información posible sobre el problema que está tratando de resolver y el apoyo que cree que necesita."
          />
          <ListedFaq
            question="¿Los voluntarios deberán convertirse en empleados del gobierno?"
            answer="No. Los equipos de gobierno son los encargados de definir las actividades y el acceso que los voluntarios tienen."
          />
          <ListedFaq
            question="¿Puedo solicitar un equipo completo?"
            answer="Si, dependerá de los voluntarios disponibles."
          />
          <ListedFaq
            question="¿Cuándo conseguiremos un voluntario?"
            answer="Después de solicitar voluntarios, nuestro equipo solicitará más información para determinar sus necesidades. Trabajaremos lo más rápido posible para asignar un equipo que se ajuste a sus proyectos y necesidades"
          />
          <ListedFaq
            question="¿Necesitamos pagar voluntarios o proporcionar recursos?"
            answer="No. Se les pide a los voluntarios que ofrezcan su tiempo como voluntarios y deben tener el equipo básico y los recursos necesarios para ayudarlos con sus tareas. Para conocer más sobre qué esperamos de nuestros voluntarios y los gobiernos que solicitan puede revisar nuestro manifiesto."
          />
          <ListedFaq
            question="¿Quién gestiona el proyecto?"
            answer="Usted y el voluntario son responsables de administrar el proyecto. Lo conectaremos con voluntarios calificados, pero el éxito del proyecto depende de su colaboración directa."
          />
        </div>
      </PageSection>
      <PageSection theme="dark" classes="padding-top-three padding-bottom-three">
        <h1 className="cormorant big">Voluntarios</h1>
        <div className="margin-top-one padding-left-one">
          <ListedFaq
            question="¿Es esta una oportunidad pagada?"
            answer="No. Esta es una oportunidad de voluntariado. Estamos pidiendo profesionales con experiencia que puedan ofrecer su tiempo como voluntarios en este esfuerzo. Para conocer más sobre qué esperamos de nuestros voluntarios y los gobiernos que solicitan puede revisar nuestro manifiesto."
          />
          <ListedFaq
            question="¿Estoy empleado por mi organización de colocación potencial?"
            answer="No. Esta es una oportunidad de voluntariado. Su condición de voluntario no es un paso hacia la obtención de empleo. Para conocer más sobre qué esperamos de nuestros voluntarios y los gobiernos que solicitan puede revisar nuestro manifiesto."
          />
          <ListedFaq
            question="¿Qué debe hacer un voluntario si algo sale mal con el proyecto?"
            answer="Comunicate con los miembros del equipo  a través de los canales establecidos. La mayoría de nuestra comunicación la hacemos a través de Slack."
          />
          <ListedFaq
            question="¿Quién verá mis datos?"
            answer="Sus datos solo se compartirán con los miembros del equipo y los socios del proyecto cuando sea necesario para facilitar la colocación. Lea más en nuestra Política de Privacidad de Datos."
          />
          <ListedFaq
            question="¿Tengo garantizada una colocación?"
            answer="No. Las necesidades de los gobiernos pueden ser muy específicas y variadas. Te pedimos que seas paciente.  Estamos evaluando cuidadosamente las necesidades de los gobiernos locales y uniendo a los solicitantes con los gobiernos locales en función de la experiencia y la habilidad."
          />
          <ListedFaq
            question="¿Puedo ser voluntario con todo mi equipo, organización o empresa?"
            answer="Si. Cuando estés llenando el formulario, en la parte de comentarios, proporciona información de tu equipo."
          />
          <ListedFaq
            question="¿Debo tomar un permiso de mi trabajo actual?"
            answer="Este es un trabajo de voluntariado. Te pedimos que nos des a conocer tu disponibilidad para poder añadirte a uno de nuestros equipos."
          />
          <ListedFaq
            question="¿Cómo examina la Brigada Digital Mx a los voluntarios?"
            answer="Los voluntarios envían una solicitud que es revisada por el equipo de Brigada Digital Mx. Si creemos que el solicitante podría ser una buena opción para un proyecto, se le asigna un equipo y un líder de proyecto. La asignación del proyecto depende de la urgencia e importancia de nuestros proyectos y de la experiencia y preparación necesaria de cada proyecto. "
          />
          <ListedFaq
            question="¿Quién gestiona el proyecto?"
            answer="El voluntario y el enlace de gobierno son los  responsables de administrar el proyecto. Conectamos voluntarios con gobiernos que necesitan ayuda, pero el éxito del proyecto depende de su colaboración directa. Brigada Digital MX hará lo posible para apoyar los proyectos establecidos."
          />
          <ListedFaq
            question="¿Donde trabajare?"
            answer="Si encontramos a un proyecto que necesite tus características, serás asignado a un proyecto y  trabajarás con una agencia gubernamental estatal o local de manera digital y remota."
          />
          <ListedFaq
            question="¿Cuánto durarán los proyectos?"
            answer="Nuestros proyectos tienen diferente naturaleza y variaran en sus tiempos."
          />
        </div>
      </PageSection>
      <Footer theme="light" classes="padding-top-three" />
    </main>
  );
};
