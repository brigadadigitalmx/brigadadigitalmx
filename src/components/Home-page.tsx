import React, { useEffect, useRef, useState } from 'react';
import { Link } from 'react-router-dom';

import { Header, Footer } from './common';
import { PageSection } from './sections';

export const HomePage = () => {
  const [collapseHeader, setCollapseHeader] = useState(false);

  useEffect(() => {
    window.addEventListener(
      'scroll',
      e => {
        if (document.getElementById('main').scrollTop > 20) {
          setCollapseHeader(true);
        }

        if (document.getElementById('main').scrollTop < 20) {
          setCollapseHeader(false);
        }
      },
      true
    );
  }, []);

  return (
    <main>
      <Header collapse={collapseHeader} />
      <PageSection theme="light" classes="padding-top-three padding-bottom-three">
        <h1 className="cormorant big">
          COVID-19 representa un reto al que nuestra sociedad pocas veces se ha enfrentado y creemos que las herramientas
          digitales pueden contribuir para ayudar y combatir esta pandemia.
        </h1>
        <p className="big margin-top-three">
          Estamos ayudando a entidades de gobierno federal, estatal y municipal durante la crisis a través de voluntarios 
          digitales con experiencia en procesamiento de datos, tecnología, comunicaciones y operaciones. 
        </p>
        <div className="buttons-list margin-top-three">
          <a href="https://forms.gle/QPkMBTPEuE4WzTkv7" className="button" target="_blank">
            Gobiernos
          </a>
          <a href="https://forms.gle/pEyMsQ7SPT6z6A2V9" className="button" target="_blank">
            Voluntarios
          </a>
        </div>
      </PageSection>
      <PageSection theme="dark" classes="padding-top-three padding-bottom-three">
        <h1 className="cormorant big">Gobiernos</h1>
        <p className="big margin-top-three">
         Si eres parte de alguna dependencia de gobierno que necesita ayuda para combatir la crisis ocasionada por el COVID-19,
          estamos construyendo una comunidad de voluntarios. Las capacidades de los voluntarios está enfocada en tecnología,
          comunicaciones y operaciones. ¡Podemos ayudar! 
        </p>
        <p className="big margin-top-one">
          Para pode ayudar necesitamos conocer tus necesidades, llena el siguiente formato (no toma más de 10 minutos) y posteriormente estaremos en contacto. 
        </p>
        <div className="buttons-list margin-top-three">
          <a href="https://forms.gle/QPkMBTPEuE4WzTkv7" className="button" target="_blank">
            Solicita Ayuda
          </a>
        </div>
      </PageSection>
      <PageSection theme="light" classes="padding-top-three padding-bottom-three">
        <h1 className="cormorant big">Voluntarios</h1>
        <p className="big margin-top-three">
          Si quieres aportar tu conocimiento, tiempo y esfuerzo ¡Estamos buscando voluntarios!
        </p>
        <p className="big">Necesitamos usuarios con conocimiento en las siguientes áreas:</p>
        <h2 className="margin-top-two">Tecnología</h2>
        <p className="big margin-top-one">
          Científicos de datos, ingenieros de front-end y back-end, diseñadores, gerentes de ingeniería,
          gerentes de producto, investigadores de usuarios, escritores.
        </p>
        <h2 className="margin-top-two">Políticas Públicas</h2>
        <p className="big margin-top-one">Licenciados/ Maestros en relaciones internacionales, ciencia política,
          economistas, abogados.
        </p>
        <h2 className="margin-top-two">Comunicaciones</h2>
        <p className="big margin-top-one">
          Estrategas de contenido, redactores, diseñadores gráficos y gerentes de redes sociales.
        </p>
        <div className="buttons-list margin-top-three">
          <a href="https://forms.gle/pEyMsQ7SPT6z6A2V9" className="button" target="_blank">
            Inscríbete
          </a>
        </div>
      </PageSection>
      <Footer theme="dark" classes="padding-top-three" />
    </main>
  );
};
