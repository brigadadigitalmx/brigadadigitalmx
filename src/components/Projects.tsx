import React from 'react';

import { Header, Footer } from './common';
import { PageSection } from './sections';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTwitter } from '@fortawesome/free-brands-svg-icons';

export const Projects = () => {
  return (
    <main>
      <Header appearance="extra-collapsed" />
      <PageSection theme="light" classes="padding-top-three padding-bottom-three">
        <h1 className="cormorant big">Proyectos</h1>
        <div className="margin-top-one padding-left-one">
          <p>Estaremos publicando sobre nuestros proyectos conforme vayan avanzando.</p>
          <p className="margin-top-one">
            Si quieres conocer más acerce de nosotros síguenos en{' '}
            <a href="#" target="_blank">
              @Brigadas_DigMX x <FontAwesomeIcon icon={faTwitter} />
            </a>
          </p>
        </div>
      </PageSection>
      <Footer theme="dark" classes="padding-top-three" />
    </main>
  );
};
