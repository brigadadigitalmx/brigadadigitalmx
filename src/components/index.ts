export * from './Home-page';
export * from './Faqs';
export * from './Us';
export * from './Projects';
