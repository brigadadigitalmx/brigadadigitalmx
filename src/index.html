<!DOCTYPE html>
<html>
    <head profile="">
        <meta charset="UTF-8" />
        <title>Brigada Digial MX</title>
        <link rel="icon" type="image/jpg" href="https://firebasestorage.googleapis.com/v0/b/brigadadigitalmx-public.appspot.com/o/favicon.png?alt=media&token=adf47698-ced7-4f9c-a0ee-17fc60850307">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta name="Description" CONTENT="Sitio web de las brigadas digitales en México">
        <meta name="google" content="notranslate" />
        <meta property="og:title" content="Brigada Digital MX">
        <meta property="og:description" content="Sitio web de las brigadas digitales en México.">
        <meta property="og:url" content="http://brigadadigitalmx.org/">
        <meta property="og:image" content="">
        <link href="https://fonts.googleapis.com/css?family=Cormorant+Garamond|Anton|Sen&display=swap" rel="stylesheet">
    </head>

    <body>
        <div id="main"></div>
    </body>
</html>
