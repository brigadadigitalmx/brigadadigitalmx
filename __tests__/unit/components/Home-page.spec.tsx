import * as React from 'react';
import { HomePage } from '../../../src/components';
import { render } from 'enzyme';

describe ('HomePage', () => {

  it('renders correctly', () => {
    const homepage = render(<HomePage />);
    expect(homepage).toMatchSnapshot();
  });
});

