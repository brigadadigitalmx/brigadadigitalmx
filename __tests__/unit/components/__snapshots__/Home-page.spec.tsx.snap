// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<main>
  <header
    class=" "
  >
    <div
      class="grid content collapse"
    >
      <div
        class="col  "
      >
        <a
          href="/"
        >
          <img
            class="logo"
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        class="col  two-spaces"
      >
        <nav>
          <a
            href="/projects"
          >
            Proyectos
          </a>
          <a
            href="/us"
          >
            Nosotros
          </a>
          <a
            href="/faqs"
          >
            FAQs
          </a>
          <a
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-twitter fa-w-16 "
              data-icon="twitter"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                fill="currentColor"
              />
            </svg>
          </a>
        </nav>
      </div>
    </div>
  </header>
  <section
    class="page-section light padding-top-three padding-bottom-three"
  >
    <h1
      class="cormorant big"
    >
      COVID-19 representa un reto de organización y servicios para los gobiernos, y creemos usar herramientas digitales para ayudar.
    </h1>
    <p
      class="big margin-top-three"
    >
      Estamos ayudando a entidades de gobierno federal, estatal y municipal durante la crisis de salud pública ocasionada por COVID-19, a través de voluntarios digitales con experiencia en procesamiento de datos, tecnología, comunicaciones y operaciones.
    </p>
    <div
      class="buttons-list margin-top-three"
    >
      <a
        class="button"
        href="https://forms.gle/cMXrYdXGxQo6QSrw9"
        target="_blank"
      >
        Gobiernos
      </a>
      <a
        class="button"
        href="https://forms.gle/3CUo2tuomGHXU6yg9"
        target="_blank"
      >
        Voluntarios
      </a>
    </div>
  </section>
  <section
    class="page-section dark padding-top-three padding-bottom-three"
  >
    <h1
      class="cormorant big"
    >
      Gobiernos
    </h1>
    <p
      class="big margin-top-three"
    >
      Si trabaja en alguna oficina de gobierno y necesita ayuda para superar la crisis de COVID-19, estamos armando una red de voluntarios capacitados en procesamiento de datos, tecnología, comunicaciones y operaciones que puede ayudarlo.
    </p>
    <p
      class="big margin-top-one"
    >
      Llene el siguiente formato y nos pondremos en contacto para analizar sus necesidades y encontrar voluntarios que puedan ayudarle:
    </p>
    <div
      class="buttons-list margin-top-three"
    >
      <a
        class="button"
        href="https://forms.gle/cMXrYdXGxQo6QSrw9"
        target="_blank"
      >
        Gobiernos
      </a>
    </div>
  </section>
  <section
    class="page-section light padding-top-three padding-bottom-three"
  >
    <h1
      class="cormorant big"
    >
      Voluntarios
    </h1>
    <p
      class="big margin-top-three"
    >
      Si quieres aportar tu conocimiento y tiempo, estamos aceptando voluntarios con conocimientos y experiencia técnica que los gobiernos necesitan.
    </p>
    <p
      class="big"
    >
      Necesitamos usuarios con conocimiento en las siguientes áreas:
    </p>
    <h2
      class="margin-top-two"
    >
      Tecnología
    </h2>
    <p
      class="big margin-top-one"
    >
      Científicos de datos, ingenieros de front-end y back-end, diseñadores, gerentes de ingeniería, gerentes de producto, investigadores de usuarios, escritores.
    </p>
    <h2
      class="margin-top-two"
    >
      Políticas Públicas
    </h2>
    <p
      class="big margin-top-one"
    >
      Políticas públicas, Epidemiólogos.
    </p>
    <h2
      class="margin-top-two"
    >
      Comunicaciones
    </h2>
    <p
      class="big margin-top-one"
    >
      Estrategas de contenido, redactores, diseñadores gráficos y gerentes de redes sociales.
    </p>
    <div
      class="buttons-list margin-top-three"
    >
      <a
        class="button"
        href="https://forms.gle/cMXrYdXGxQo6QSrw9"
        target="_blank"
      >
        Voluntarios
      </a>
    </div>
  </section>
  <footer
    class="padding-top-three dark"
  >
    <h1>
      Contactanos
    </h1>
    <p>
      contacto@brigadadigitalmx.org
    </p>
  </footer>
</main>
`;
